internal/util/cmdutil/cmdutil.go

```diff
@@ -113,3 +118,51 @@ func StringToImagePullPolicy(v string) fnruntime.ImagePullPolicy {
 		return fnruntime.AlwaysPull
 	}
 }
+
+// WriteFnOutput writes the output resources of function commands to provided destination
+func WriteFnOutput(dest, content string, fromStdin bool, w io.Writer) error {
+	r := strings.NewReader(content)
+	switch dest {
+	case StdOut:
+		// if user specified dest is "stdout" directly write the content as it is already wrapped
+		_, err := w.Write([]byte(content))
+		return err
+	case Unwrap:
+		// if user specified dest is "unwrap", write the unwrapped content to the provided writer
+		return WriteToOutput(r, w, "")
+	case "":
+		if fromStdin {
+			// if user didn't specify dest, and if input is from STDIN, write the wrapped content provided writer
+			// this is same as "stdout" input above
+			_, err := w.Write([]byte(content))
+			return err
+		}
+	default:
+		// this means user specified a directory as dest, write the content to dest directory
+		return WriteToOutput(r, nil, dest)
+	}
+	return nil
+}
+
+// WriteToOutput reads the input from r and writes the output to either w or outDir
+func WriteToOutput(r io.Reader, w io.Writer, outDir string) error {
+	var outputs []kio.Writer
+	if outDir != "" {
+		if _, err := os.Stat(outDir); err != nil {
+			if errors.Is(err, os.ErrNotExist) {
```

Comment:
If we decide about this behavior (comment above), then we should perform this check in the beginning of the `render or eval`.

Reply:
We should not require directory to exist. This behavior also needs to be specified in reference docs.

Reply:
Yes. Updated.